services/consensusalgo/leanhelixconsensus/test/metrics_test.go

```diff
@@ -0,0 +1,127 @@
+	"github.com/orbs-network/lean-helix-go/services/randomseed"
+	"github.com/orbs-network/lean-helix-go/spec/types/go/protocol"
+	"github.com/orbs-network/orbs-network-go/test"
+	"github.com/orbs-network/orbs-network-go/test/builders"
+	"github.com/orbs-network/orbs-network-go/test/crypto/keys"
+	"github.com/orbs-network/orbs-network-go/test/with"
+	"github.com/stretchr/testify/require"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestMetricsAreUpdatedOnElectionTrigger(t *testing.T) {
+
+	with.Concurrency(t, func(ctx context.Context, parent *with.ConcurrencyHarness) {
+		h := newLeanHelixServiceHarness(0, 10*time.Millisecond).start(parent, ctx)
+
+		h.beFirstInCommittee() // just so RequestOrderedCommittee will succeed
+		h.expectGossipSendLeanHelixMessage()
+
+		h.handleBlockSync(ctx, 1)
+
+		// Election trigger should fire and metrics should be updated
+
+		metrics := h.getMetrics()
+
+		require.True(t, test.Eventually(1*time.Second, func() bool {
+			return metrics.currentElectionCount.Value() > 0
+		}), "expected currentElectionCount metric to update")
+
+		require.True(t, test.Eventually(1*time.Second, func() bool {
+			return metrics.currentLeaderMemberId.Value() != ""
+		}), "expected currentLeaderMemberId metric to update")
+
+		require.True(t, test.Eventually(1*time.Second, func() bool {
+			matched, err := regexp.MatchString("samples=[1-9]", metrics.timeSinceLastElectionMillis.String())
+			if err != nil {
+				panic(err)
+			}
+			return matched
+		}), "expected timeSinceLastElectionMillis metric to update")
+	})
+
+}
+
+func TestMetricsAreUpdatedOnCommit(t *testing.T) {
+
+	with.Concurrency(t, func(ctx context.Context, parent *with.ConcurrencyHarness) {
+		h := newLeanHelixServiceHarness(0, time.Hour).start(parent, ctx)
+
+		h.expectGossipSendLeanHelixMessage()
+		h.expectValidateTransactionBlock()
+		h.expectValidateResultsBlock()
+		h.expectCommitBlock()
+
+		h.dontBeFirstInCommitee()
+
+		const syncedBlockHeight = 1
+		h.handleBlockSync(ctx, syncedBlockHeight)
+
+		orderedCommittee := h.requestOrderingCommittee(ctx)
+		orderedComitteeNodeIndicies := keys.NodeAddressesForTestsToIndexes(orderedCommittee.NodeAddresses)
+
+		leaderNodeIndex := orderedComitteeNodeIndicies[0] // leader for view 0 is always the first member in the ordered committee
+
+		prevBlockProof := protocol.BlockProofReader(nil) // the previous block (from sync) had no proof
+		randomSeed := randomseed.CalculateRandomSeed(prevBlockProof.RandomSeedSignature())
+
+		const firstTermHeight = syncedBlockHeight + 1
+		const view = 0
+		blockPair := builders.BlockPair().WithHeight(firstTermHeight).WithTransactions(1).WithEmptyLeanHelixBlockProof().Build()
+
+		h.handlePreprepareMessage(ctx, blockPair, firstTermHeight, view, leaderNodeIndex)
+
+		commitMessages := []*interfaces.CommitMessage{}
+		for i := 0; i < h.networkSize(); i++ {
+			cmsg := h.handleCommitMessage(ctx, blockPair, firstTermHeight, view, randomSeed, i)
```

Comment:
protocol question - are we allowed to commit a block without receiving a single prepare message for it? I don't know if there is a safety issue here, since 2f commit messages will only arrive after sufficient prepare messages were sent in the real network with f +1 honest nodes. but maybe ask @gadcl if there are no view change scenarios where this may be dangerous?

Reply:
Good question. I also don't think there is a safety issue here, but will ask Gad. I think we can push this PR regardless, because if there's an issue here it would require an unrelated LH fix.